### DISCUSSION TITLE
"gazebo:= Ignition"

### DISCUSSION THREAD

**Discussion by Desolate-Tears** [turtlebot/turtlebot4#500] [Q&A]
regard robot_description.launch.py file
In the robot_state_publisher node, "gazebo:=ignition" in the robot_description parameter.    I had no idea what it meant until I looked up the gazebo official documentation and found that the gazebo community had announced that it would retire the Ignition name and replace it with the Gazebo name, but the source file still had "gazebo:=ignition" in it.   I did not find the Ignition specific configuration in the xacro source file.   If ignition is not needed, and ignition is no longer used, can I comment it out?   .   This particular configuration bothers me. Thank you very much

**Comment by slowrunner** [comment]
I found why that value is there!  In the create3_sim/irobot_create_description/urdf/create3.urdf.xacro:

```
 <xacro:if value="${'$(arg gazebo)' == 'ignition'}">
    <xacro:property name="body_cog_x_offset"       value="${2.28*cm2m}" />
  </xacro:if>
  <xacro:if value="${'$(arg gazebo)' == 'classic'}">
    <xacro:property name="body_cog_x_offset"       value="${0*cm2m}" />
  </xacro:if>
```
and
```
 <xacro:if value="${'$(arg gazebo)' == 'ignition'}">
    <xacro:property name="caster_position_z"       value="${-5.03*cm2m}" />
  </xacro:if>
  <xacro:if value="${'$(arg gazebo)' == 'classic'}">
    <xacro:property name="caster_position_z"       value="${-5.2*cm2m}" />
  </xacro:if>
```

Apparently the two simulators need different z offsets
